docs/apigateway.md

```diff
@@ -1,53 +1,105 @@
-```
-A new URL is generated exposing the `echo` action via a __GET__ HTTP method.
+## Create APIs from OpenWhisk web actions using the CLI
 
-Let's give it a try by sending a HTTP request to the URL.
-```
-$ curl https://21ef035.api-gw.mybluemix.net/hello/echo?marco=polo
-```
-This will invoke the `echo` action, returning back a JSON string with the parameters sent.
-```
-{
-  "marco":"polo"
-}
-```
+### OpenWhisk CLI configuration
 
-You can pass parameters to the action via simple query parameters, or via request body.
+Follow the instructions in [Configure CLI](./README.md#setting-up-the-openwhisk-cli) on how to set the authentication key for your specific namespace.
 
-### Exposing multiple actions
+**Note:** The APIs you created using the `wsk api-experimental` will continue to work for a short period, however you should begin migrating your APIs to web actions and reconfigure your existing apis using the new CLI command `wsk api`.
+
+### Create your first API using the CLI
+
+1. Create a JavaScript file with the following content. For this example, the file name is 'hello.js'.
+  ```javascript
+  function main({name:name='Serverless API'}) {
+      return {payload: `Hello world ${name}`};
+  }
+  ```
+  
+2. Create a web action from the following JavaScript function. For this example, the action is called 'hello'. Make sure to add the flag `--web true`
+  
+  ```
+  wsk action create hello hello.js --web true
+  ```
+  ```
+  ok: created action hello
+  ```
+  
+3. Create an API with base path `/hello`, path `/world` and method `get` with response type `json`
+  
+  ```
+  wsk api create /hello /world get hello --response-type json
+  ```
+  ```
+  ok: created API /hello/world GET for action /_/hello
+  https://${APIHOST}:9001/api/21ef035/hello/world
+  ```
+  A new URL is generated exposing the `hello` action via a __GET__ HTTP method.
+  
+4. Let's give it a try by sending a HTTP request to the URL.
+  
+  ```
+  $ curl https://${APIHOST}:9001/api/21ef035/hello/world?name=OpenWhisk
+  ```
+  ```json
+  {
+  "payload": "Hello world OpenWhisk"
+  }
+  ```
+  The action `hello` got invoked, returning back a JSON string including the parameter `name` sent via query parameter. You can pass parameters to the action via simple query parameters, or via request body.
+  
+### Full control over the HTTP response
+  
+  The `--response-type` flag controls the target url of the web action to be proxied by the API Gateway. Using `--response-type json` as above returns the full result of the action in JSON format and automatically sets the Content-Type header to `application/json` which enables you to easily get started. 
+  
+  Once you get started you want to have full control over the http response properties like `statusCode`, `headers` and return different content types in the `body`. You can do this by using `--response-type http`, this will configure the target url of the web action with the `http` extension.
```

Comment:
I think I want lower case `http` for the value of --response-type and when referring to the extension, other than that I agree HTTP, I will review.